### PR TITLE
Add test showing incorrect sequence point placement

### PR DESCRIPTION
**Pull request by tmat**
Issue: https://github.com/dotnet/roslyn/issues/32352

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs [hunk 1]
```diff
@@ -2100,5 +2100,77 @@ static int F(int x, out int p)
...
+        {
+            var inner = false;
+            var f = new Func<bool>(() => inner & outer);
+        }
+    }
+}");
+            var c = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20, options: TestOptions.DebugDll);
+
+            // TODO: https://github.com/dotnet/roslyn/issues/32352
+            // The inner closure allocation on IL_0018 should be withing the sequence point associated with opening brace of the if statement.
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs:2125]: Typo: within
```suggestion
            // The inner closure allocation on IL_0018 should be within the sequence point associated with opening brace of the if statement.
```

File: src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs [hunk 2]
```diff
@@ -2100,5 +2100,77 @@ static int F(int x, out int p)
...
+    void F(bool outer)
+    {
+        if (outer) 
+        {
+            var inner = false;
+            var f = new Func<bool>(() => inner & outer);
+        }
+    }
+}");
+            var c = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20, options: TestOptions.DebugDll);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs:2122]: nit: the `targetFramework` and `options` could be removed

File: src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs [hunk 3]
```diff
@@ -2100,5 +2100,77 @@ static int F(int x, out int p)
       </methods>
     </symbols>", format: CodeAnalysis.Emit.DebugInformationFormat.Pdb);
         }
+
+        [Fact]
+        [WorkItem(32352, "https://github.com/dotnet/roslyn/issues/32352")]
+        public void ClosureAllocationSequencePoints()
+        {
+            var source = WithWindowsLineBreaks(@"
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs:2108]: >WithWindowsLineBreaks [](http://example.com/codeflow?start=25&length=21)

nit: I wouldn't expect this `WithWindowsLineBreaks` to be necessary

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2) with minor nits to consider

**Comment by AlekseyTs** [comment]
@tmat Just an FYI, this is a test-only PR and can be merged with a single sign-off from compiler team

**Pull request merged**